gravityflow/gravityflow
# Add the step merge tag attribute

This PR does the following:

1. Adds the step merge tag attribute to allow merge tags to specify the step for which tokens must be generated. This allows feed add-ons to specify the step. For example, a Twilio or Slack message can contain a one-click approval link for the next step.

2. Fixes an edge case where the assignee attribute will remove the assignee added in the constructor and override the assignee in subsequent instances of the merge tag.

3. Fixes an issue with the current version of the Twilio Add-On where URLs get encoded breaking parameters. There's another issue with the shortener - currently, the URL Shortener in the Twilio Add-On will break links with more than one parameter. This is addressed in the Twilio Add-On repo PR5.

Fixes [HS#6406](https://secure.helpscout.net/conversation/624596724/6406?folderId=516732)

**Testing Instructions**
1. Add a Twilio feed with the one-click approval merge tag for an approval step approval step. e.g {workflow_approve_url: assignee="assignee_field|4" step="300" token="true"} where the assignee attribute is the assignee key (e.g. assignee_user_field|4 or email_field|4 ) and step is the step ID of the approval step.
2. On master, the URL sent via Twilio will be broken. On this branch, the SMS will contain the URL intact.
3. With PR5 in the Twilio Add-On repo, the shortener will shorten the link correctly.
4. Check for regression issues.
